src/dashboard/src/media/js/archival_storage/archival_storage_search.js

```diff
@@ -101,33 +100,314 @@ $(document).ready(function() {
   // Ensure the select field name field is hidden/displayed as appropriate
   selectField($('.advanced_search_form_field_field > select')[0]);
 
-  function aipSearchSubmit() {
-    var destination = '/archival-storage/search/' + '?' + search.toUrlParams();
-    if($('#id_show_files').is(':checked')) {
-      destination += '&filemode=true';
+  if (on_success !== null) {
+    function aipSearchSubmit() {
+      // Query Django, which queries ElasticSearch, to get the backlog file info
+      var query_url = search_uri + '?' + search.toUrlParams();
+      if($('#id_show_files').is(':checked')) {
+        query_url += '&filemode=true';
+      }
+      $.ajax({
+        type: 'GET',
+        url: query_url,
+        data: null,
+        success: on_success,
+        error: function(jqXHR, textStatus, errorThrown) {
+          on_error(query_url, jqXHR, textStatus, errorThrown);
+        },
+      });
     }
-    if($('#id_show_aics').is(':checked')) {
-      destination += '&show_aics=true';
+
+    // submit logic
+    $('#search_submit').click(function() {
+      aipSearchSubmit();
+    });
+
+    $('#search_form').submit(function() {
+      aipSearchSubmit();
+      return false;
+    });
+  }
+
+  return search;
+}
+
+$(document).ready(function() {
+
+  set_create_aic_visibility();
```

Comment:
One small change in behavior I notice is that the create AIC button will show now without performing a search. I personally don't think that's a bad thing. Having to set its display if a search is performed is extra logic to deal with. That being said, it's not my decision! :p And it does point to the logic for the set visibility function not working as anticipated. So it probably does need looking at if you can also confirm that behavior for yourself.

Reply:
> And it does point to the logic for the set visibility function not working as anticipated. So it probably does need looking at if you can also confirm that behavior for yourself.

Leaving aside the question of whether the Create AIC button should show before the user runs a search, is inconsistent visibility that you noticed in your testing? In my testing, the set visibility function worked fine once I remembered to call the function when the document is ready in addition to when search results are refreshed.

Reply:
On the larger question of whether the button should show on page refresh, I had a conversation with @sromkey @ablwr and the decision was to show the button regardless of whether a user has run a search. Rationale: "In terms of UX/design, we should go with whatever causes the least amount of surprise for the user, and I think in general an initially-hidden button that you have to uncover through searching is a 'surprise'".

Based on that conversation, I will also add a tooltip to the "Create an AIC" button that clarifies that clicking on it will create an AIC from the currently displayed search results.